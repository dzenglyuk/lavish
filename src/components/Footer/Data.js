export const infoLinks = [
    {
        title: "About Us",
        links: [
            {
                name: "How it works",
                value: "/sign-up",
            },
            {
                name: "Testimonials",
                value: "/",
            },
            {
                name: "Careers",
                value: "/",
            },
            {
                name: "Investors",
                value: "/",
            },
            {
                name: "Terms of Service",
                value: "/",
            },
        ],
    },
    {
        title: "Contact Us",
        links: [
            {
                name: "Contact",
                value: "/",
            },
            {
                name: "Support",
                value: "/",
            },
            {
                name: "Destinations",
                value: "/",
            },
            {
                name: "Sponsorships",
                value: "/",
            },
        ],
    },
];

export const socialLinks = [
    {
        title: "Videos",
        links: [
            {
                name: "Submit Video",
                value: "/",
            },
            {
                name: "Ambassadors",
                value: "/",
            },
            {
                name: "Agency",
                value: "/",
            },
            {
                name: "Influencer",
                value: "/",
            },
        ],
    },
    {
        title: "Social Media",
        links: [
            {
                name: "Instagram",
                value: "/",
            },
            {
                name: "Facebook",
                value: "/",
            },
            {
                name: "Youtube",
                value: "/",
            },
            {
                name: "Twitter",
                value: "/",
            },
        ],
    },
];

export const subscription = {
    heading: "Join our exclusive membership to receive the latest news and trends",
    text: "You can unsubscribe at any time."
};