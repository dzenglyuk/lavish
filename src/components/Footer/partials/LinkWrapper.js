import React from "react";
import { Link } from "react-router-dom";

const LinkWrapper = ({ linkObjects }) => {
    return (
        <div className="footer-link-wrapper">
            {linkObjects.map((linkObject) => (
                <div className="footer-link-items" key={linkObject.title}>
                    <h2>{linkObject.title}</h2>
                    {linkObject.links.map((link) => (
                        <Link to={link.value} key={link.name}>
                            {link.name}
                        </Link>
                    ))}
                </div>
            ))}
        </div>
    );
};

LinkWrapper.propTypes = {
    linkObjects: (props, propName, componentName) => {
        const propValue = (props || {})[propName];
        const isNotArray = (value) => !Array.isArray(value);

        if (propValue === undefined)
            return new Error(
                `${propName} of component ${componentName} is required`
            );

        if (isNotArray(propValue))
            return new Error(
                `${propName} of component ${componentName} must be array`
            );

        if (typeof propValue[0] !== "object")
            return new Error(
                `${propName} of component ${componentName} must be array of objects`
            );

        return null;
    }
};

export default LinkWrapper;
