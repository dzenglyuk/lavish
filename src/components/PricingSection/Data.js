const starterAccount = {
    info: {
        name: "Starter",
        price: "$8.99",
        period: "per month"
    },
    features: ["100 Transactions", "2% Cash Back", "$10,000 Limit"]
};

const goldAccount = {
    info: {
        name: "Gold",
        price: "$29.99",
        period: "per month"
    },
    features: ["1000 Transactions", "3.5% Cash Back", "$100,000 Limit"]
};

const diamondAccount = {
    info: {
        name: "Diamond",
        price: "$99.99",
        period: "per month"
    },
    features: ["Unlimited Transactions", "5% Cash Back", "Unlimited Spending"]
};

export {
    starterAccount,
    goldAccount,
    diamondAccount
};