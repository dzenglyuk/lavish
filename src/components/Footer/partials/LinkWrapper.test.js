import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { StaticRouter as Router } from "react-router";

import { infoLinks, socialLinks } from "../Data";
import LinkWrapper from "./LinkWrapper";

test("Renders correctly", () => {
    const { getByText } = render(<Router><LinkWrapper linkObjects={infoLinks}/></Router>);
    const names = infoLinks[0].links.map(link => link.name);
    getByText(infoLinks[0].title);
    getByText(infoLinks[1].title);
    for (let name of names) {
        getByText(name);
    }
});

test("Matches the snapshot 1", () => {
    const tree = renderer.create(<Router><LinkWrapper linkObjects={infoLinks}/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 2", () => {
    const tree = renderer.create(<Router><LinkWrapper linkObjects={socialLinks}/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});
