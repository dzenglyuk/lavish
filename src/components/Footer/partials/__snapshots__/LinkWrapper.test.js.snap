// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1 1`] = `
<div
  className="footer-link-wrapper"
>
  <div
    className="footer-link-items"
  >
    <h2>
      About Us
    </h2>
    <a
      href="/sign-up"
      onClick={[Function]}
    >
      How it works
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Testimonials
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Careers
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Investors
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Terms of Service
    </a>
  </div>
  <div
    className="footer-link-items"
  >
    <h2>
      Contact Us
    </h2>
    <a
      href="/"
      onClick={[Function]}
    >
      Contact
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Support
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Destinations
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Sponsorships
    </a>
  </div>
</div>
`;

exports[`Matches the snapshot 2 1`] = `
<div
  className="footer-link-wrapper"
>
  <div
    className="footer-link-items"
  >
    <h2>
      Videos
    </h2>
    <a
      href="/"
      onClick={[Function]}
    >
      Submit Video
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Ambassadors
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Agency
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Influencer
    </a>
  </div>
  <div
    className="footer-link-items"
  >
    <h2>
      Social Media
    </h2>
    <a
      href="/"
      onClick={[Function]}
    >
      Instagram
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Facebook
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Youtube
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      Twitter
    </a>
  </div>
</div>
`;
